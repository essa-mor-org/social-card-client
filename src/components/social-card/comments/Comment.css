.Comment_panel{
    display: flex;
    padding-left: var(--Social-space);
    padding-bottom: var(--Social-space);
}

.Comment_comment_panel{
    background-color: var(--Social-bg-color);
    border-radius: calc(var(--Social_comment_radius)/2);
    max-width:  var(--Social-comment-width);
    padding: 10px 20px 10px 20px;
    box-sizing: border-box;
}

.Comment_profile{
    color: var(--Social-color-link);
    font-size: var(--Social-font-smallmeduim);
    text-decoration-line: none;
    font-weight: bold;
    margin-right: 5px;
}

.Comment_profile:hover{
    text-decoration-line: underline;
}

.Comment_comment{
    font-size: var(--Social-font-smallmeduim);
    color: #1d2129;
}