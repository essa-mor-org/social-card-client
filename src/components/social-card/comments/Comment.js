import React from 'react';
import PropTypes from 'prop-types';

import './Comment.css';

function Comment({profileImage ,profileName, comment}) {
	return (<div className="Comment_panel">
		<img className="Social_comments_image" src={profileImage} alt="" />
		<div className="Comment_comment_panel">
			<a className="Comment_profile" href="#">{profileName}</a>
			<span className="Comment_comment">{comment}</span>
		</div>
	</div>);
}

Comment.propTypes = {
	profileImage: PropTypes.string,
	profileName: PropTypes.string,
	comment: PropTypes.string
};
export default Comment;