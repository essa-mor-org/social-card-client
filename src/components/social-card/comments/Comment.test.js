import React from 'react';
import Comment from './Comment';
import renderer from 'react-test-renderer';

describe('Comment test', () => {
	it('renders correctly - proprties', () => {
		const tree = renderer
			.create(<Comment />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('renders correctly - proprties', () => {
		const tree = renderer
			.create(<Comment
				profileImage="https://scontent.ftlv4-1.fna.fbcdn.net/v/t1.0-1/p80x80/n.jpg"
				profileName="Vevo"
				comment="Hi there." />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
