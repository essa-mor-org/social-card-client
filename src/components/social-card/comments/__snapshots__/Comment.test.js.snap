// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment test renders correctly - no proprties 1`] = `
<div
  className="Comment_panel"
>
  <img
    alt=""
    className="Social_comments_image"
    src={undefined}
  />
  <div
    className="Comment_comment_panel"
  >
    <a
      className="Comment_profile"
      href="#"
    />
    <span
      className="Comment_comment"
    />
  </div>
</div>
`;

exports[`Comment test renders correctly - proprties 1`] = `
<div
  className="Comment_panel"
>
  <img
    alt=""
    className="Social_comments_image"
    src="https://scontent.ftlv4-1.fna.fbcdn.net/v/t1.0-1/p80x80/n.jpg"
  />
  <div
    className="Comment_comment_panel"
  >
    <a
      className="Comment_profile"
      href="#"
    >
      Vevo
    </a>
    <span
      className="Comment_comment"
    >
      Hi there.
    </span>
  </div>
</div>
`;
